core/trino-main/src/main/java/io/trino/sql/planner/RelationPlanner.java

```diff
@@ -305,26 +305,26 @@ private RelationPlan addColumnMasks(Table table, RelationPlan plan)
         PlanBuilder planBuilder = newPlanBuilder(plan, analysis, lambdaDeclarationToSymbolMap, session, plannerContext)
                 .withScope(analysis.getAccessControlScope(table), plan.getFieldMappings()); // The fields in the access control scope has the same layout as those for the table scope
 
+        Map<Symbol, Expression> assignments = new LinkedHashMap<>();
+        for (Symbol symbol : planBuilder.getRoot().getOutputSymbols()) {
+            assignments.put(symbol, symbol.toSymbolReference());
+        }
+
         for (int i = 0; i < plan.getDescriptor().getAllFieldCount(); i++) {
             Field field = plan.getDescriptor().getFieldByIndex(i);
 
             for (Expression mask : columnMasks.getOrDefault(field.getName().orElseThrow(), ImmutableList.of())) {
                 planBuilder = subqueryPlanner.handleSubqueries(planBuilder, mask, analysis.getSubqueries(mask));
-
-                Map<Symbol, Expression> assignments = new LinkedHashMap<>();
-                for (Symbol symbol : planBuilder.getRoot().getOutputSymbols()) {
-                    assignments.put(symbol, symbol.toSymbolReference());
-                }
                 assignments.put(plan.getFieldMappings().get(i), coerceIfNecessary(analysis, mask, planBuilder.rewrite(mask)));
```

Comment:
It turns out this is actually incorrect now unless `analysis.getColumnMasks(table)` returns at most a single mask per column. Otherwise, the assignments will be overwritten and the last mask will win -- all the others will silently be ignored.

Reply:
Yes, currently the way it works is the last one wins. That's what we've been discussing in this ticket and proposed a separate change to change the SPI to expect a single mask, right?

Reply:
We only talked about limiting multiple masks across multiple connectors in this PR and changing the SPI to constrain it at that level in a separate change. However, without limiting multiple masks regardless of where they come from (single connector vs multiple connectors), this code introduces a silent correctness problem.

Reply:
I see. Should I throw an exception here if there are more than one mask, to avoid the silent last-one-wins misbehavior?
https://github.com/trinodb/trino/blob/master/core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java#L2225